--- origsrc/src/winsup/cygwin/Makefile.in	2013-02-13 10:34:03.000000000 -0600
+++ src/src/winsup/cygwin/Makefile.in	2013-02-13 15:16:23.767777400 -0600
@@ -49,11 +48,7 @@ tooldir:=$(exec_prefix)
 else
 tooldir:=$(exec_prefix)/$(target_alias)
 endif
-ifeq ($(target_alias),x86_64-pc-cygwin)
-toollibdir:=$(tooldir)/lib64
-else
 toollibdir:=$(tooldir)/lib
-endif
 else
 tooldir:=$(exec_prefix)/$(target_alias)
 toollibdir:=$(tooldir)/lib
